data_safe_haven/config/config.py

```diff
@@ -284,18 +285,17 @@ def is_complete(self, *, require_sres: bool) -> bool:
             return False
         return True
 
+    @staticmethod
+    def sanitise_sre_name(name: str) -> str:
+        return alphanumeric(name).lower()
+
     def sre(self, name: str) -> ConfigSectionSRE:
         """Return the config entry for this SRE creating it if it does not exist"""
         if name not in self.sres.keys():
-            highest_index = max(0 + sre.index for sre in self.sres.values())
-            self.sres[name] = ConfigSectionSRE(index=highest_index + 1)
+            msg = f"SRE {name} does not exist"
+            raise DataSafeHavenConfigError(msg)
         return self.sres[name]
 
-    def remove_sre(self, name: str) -> None:
-        """Remove SRE config section by name"""
-        if name in self.sres.keys():
-            del self.sres[name]
-
```

Comment:
This is called by the `teardown sre` command.

Reply:
Hmm, I think that is what I was intending (apart from breaking the code where it is called). Consistent with `deploy sre` not adding anything to the configuration.

(Another option would be a more, "make configuration" -> `dsh sync` workflow, but I'm not entirely convinced that is right).

I could see some utility in bringing an SRE down without deleting it's configuration. Similar to how `pulumi destroy` works. At least in testing I think I want to redeploy the same thing from scratch fairly often.

Reply:
OK, if it's deliberate then I think you just need to remove the call from `teardown sre`.

Reply:
Done in 6db60c4